bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/DiscoverComponents.java

```diff
@@ -93,11 +95,12 @@ public void processMessage(String topic, byte[] payload) {
         HaID haID = new HaID(topic);
         String config = new String(payload);
 
+        @Nullable
         AbstractComponent<?> component = null;
```

Comment:
The compiler is more intelligent on local variables. Therefore, the Nullable could be removed. Please check all.

Reply:
I don't think so. 
I think it would be better to follow the clean rules and mark all nullable vars explicitly, without guessing what compiler mean about this particular case. Also Idea's linter showing the warning here.
This way is closer to Optional type, that I hope will be used in future.

Reply:
I know that IntelliJ generates a false warning on these. I think it's obvious that a variable is nullable when it is initialized with `null`. The openHAB code uses the nullable annotation on local variables only if it's necessary.

Reply:
`@Nullable` annotations removed from Local variables declarations. 
I reported this problem to IntelliJ. As far as I understood `@NonNullByDefault` documentation, it should never add the `@NonNull` to local variables in nested code block.

There is a weird type mapping in IDE sources:
`private static final Map<String, PsiAnnotation.TargetType> TARGET_MAP =
    EntryStream.of("FIELD", PsiAnnotation.TargetType.FIELD,
                   "PARAMETER", PsiAnnotation.TargetType.PARAMETER,
                   "RETURN_TYPE", PsiAnnotation.TargetType.METHOD,
                   "TYPE_PARAMETER", PsiAnnotation.TargetType.TYPE_PARAMETER,
                   "TYPE_BOUND", PsiAnnotation.TargetType.TYPE_USE,
                   "TYPE_ARGUMENT", PsiAnnotation.TargetType.TYPE_USE).toImmutableMap();`

Reply:
FYI: the issue with IntelliJ Fixed, Available in: NEXT 2021.2 EAP 6